StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx

```diff
@@ -620,17 +620,22 @@ Double_t StFcsPulseAna::PulseFit(Int_t Start, Int_t End)
   else{ mFitSum = true; return mSumAdcFit; }//If fit failed return 0 don't do a histogram sum
 }
 
-void StFcsPulseAna::SetFitPars(TF1* func)
+void StFcsPulseAna::SetFitPars(TF1*& func, int start, int end)
```

Comment:
This argument was changed because in line 639 of that file there is a check if(func==0){create a new function}. Without the reference the pointer to this new function will be lost since SetFitPars() does not return the TF1* pointer that if may have created. This was certainly bad and now fixed but not the main issue.

Reply:
Fair enough. I haven't noticed that you actually create a new function and return it in the argument.